### PR TITLE
Fix LFCLK sources

### PR DESCRIPTION
**Pull request by nonsintetic**
LFCLK source was hard-coded to be RC oscillator, now supports all other
sources from the tools menu

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -442,6 +442,12 @@ BLENano.menu.softdevice.s130.upload.maximum_size=151552
 BLENano.menu.softdevice.s130.build.extra_flags=-DNRF51 -DS130 -DNRF51_S130
 BLENano.menu.softdevice.s130.build.ldscript=armgcc_s130_nrf51822_{build.chip}.ld
 
+BLENano.menu.lfclk.lfxo=Crystal Oscillator
```
Comment (d00616) [inline boards.txt]: I think it's simpler to add -DUSE_LFXO to extra_flags because the Crystal Oscillator is hard wired.

Comment (nonsintetic) [inline boards.txt]: I was thinking maybe some people will want to use the RC instead, if they use these as dev boards, perhaps the final circuit will not have a crystal.

File: cores/nRF5/wiring.c [hunk 2]
```diff
@@ -31,7 +31,13 @@ void init( void )
...
   NVIC_EnableIRQ(RTC1_IRQn);
 
-  NRF_CLOCK->LFCLKSRC = (uint32_t)((CLOCK_LFCLKSRC_SRC_RC << CLOCK_LFCLKSRC_SRC_Pos) & CLOCK_LFCLKSRC_SRC_Msk);
+  #if defined(USE_LFRC)
+    NRF_CLOCK->LFCLKSRC = (uint32_t)((CLOCK_LFCLKSRC_SRC_RC << CLOCK_LFCLKSRC_SRC_Pos) & CLOCK_LFCLKSRC_SRC_Msk);
+  #elif defined(USE_LFSYNT)
+    NRF_CLOCK->LFCLKSRC = (uint32_t)((CLOCK_LFCLKSRC_SRC_Synth << CLOCK_LFCLKSRC_SRC_Pos) & CLOCK_LFCLKSRC_SRC_Msk);
+  #else
+    NRF_CLOCK->LFCLKSRC = (uint32_t)((CLOCK_LFCLKSRC_SRC_Xtal << CLOCK_LFCLKSRC_SRC_Pos) & CLOCK_LFCLKSRC_SRC_Msk);
+  #endif
```
Comment (sandeepmistry) [inline cores/nRF5/wiring.c]: Can we make the default case (the else) the RC for backwards compatibility?

**Comment by sandeepmistry** [comment]
@nonsintetic ping ...

**Comment by nonsintetic** [comment]
changed it around, please check it now. sorry bout the delay, didn't see you replied

File: boards.txt [hunk 3]
```diff
@@ -442,6 +442,12 @@ BLENano.menu.softdevice.s130.upload.maximum_size=151552
...
+BLENano.menu.lfclk.lfxo.build.lfclk_flags=-DUSE_LFXO
+BLENano.menu.lfclk.lfrc=RC Oscillator
+BLENano.menu.lfclk.lfrc.build.lfclk_flags=-DUSE_LFRC
+BLENano.menu.lfclk.lfsynt=Synthesized
+BLENano.menu.lfclk.lfsynt.build.lfclk_flags=-DUSE_LFSYNT
```
Comment (sandeepmistry) [inline boards.txt]: Could you please remove this menu options for now? Later on I'd like to auto select the LFCLK source depending on if the board has it build in.

File: boards.txt [hunk 4]
```diff
@@ -524,3 +530,47 @@ Waveshare_BLE400.menu.softdevice.s130.softdeviceversion=2.0.1
 Waveshare_BLE400.menu.softdevice.s130.upload.maximum_size=151552
 Waveshare_BLE400.menu.softdevice.s130.build.extra_flags=-DNRF51 -DS130 -DNRF51_S130
 Waveshare_BLE400.menu.softdevice.s130.build.ldscript=armgcc_s130_nrf51822_{build.chip}.ld
+
+Waveshare_BLE400.menu.lfclk.lfxo=Crystal Oscillator
+Waveshare_BLE400.menu.lfclk.lfxo.build.lfclk_flags=-DUSE_LFXO
+Waveshare_BLE400.menu.lfclk.lfrc=RC Oscillator
+Waveshare_BLE400.menu.lfclk.lfrc.build.lfclk_flags=-DUSE_LFRC
+Waveshare_BLE400.menu.lfclk.lfsynt=Synthesized
+Waveshare_BLE400.menu.lfclk.lfsynt.build.lfclk_flags=-DUSE_LFSYNT
```
Comment (sandeepmistry) [inline boards.txt]: Same here.

File: boards.txt [hunk 5]
```diff
@@ -524,3 +530,47 @@ Waveshare_BLE400.menu.softdevice.s130.softdeviceversion=2.0.1
...
+ng_beacon.menu.softdevice.s110.upload.maximum_size=151552
+ng_beacon.menu.softdevice.s110.build.extra_flags=-DNRF51 -DS110 -DNRF51_S110
+ng_beacon.menu.softdevice.s110.build.ldscript=armgcc_s110_nrf51822_xxaa.ld
+
+ng_beacon.menu.softdevice.s130=S130
+ng_beacon.menu.softdevice.s130.softdevice=s130
+ng_beacon.menu.softdevice.s130.softdeviceversion=2.0.1
+ng_beacon.menu.softdevice.s130.upload.maximum_size=151552
+ng_beacon.menu.softdevice.s130.build.extra_flags=-DNRF51 -DS130 -DNRF51_S130
+ng_beacon.menu.softdevice.s130.build.ldscript=armgcc_s130_nrf51822_xxaa.ld
```
Comment (sandeepmistry) [inline boards.txt]: Could you please rebase the branch, this was added in another PR.

File: cores/nRF5/wiring.c [hunk 6]
```diff
@@ -31,7 +31,13 @@ void init( void )
   NVIC_ClearPendingIRQ(RTC1_IRQn);
   NVIC_EnableIRQ(RTC1_IRQn);
 
-  NRF_CLOCK->LFCLKSRC = (uint32_t)((CLOCK_LFCLKSRC_SRC_RC << CLOCK_LFCLKSRC_SRC_Pos) & CLOCK_LFCLKSRC_SRC_Msk);
+  #if defined(USE_LFXO)
```
Comment (sandeepmistry) [inline cores/nRF5/wiring.c]: Looks good!

File: boards.txt (hunk 5 above)
Comment (nonsintetic) [inline boards.txt]: Sorry I have no idea how to do that, I don't usually use GIT..

**Comment by sandeepmistry** [comment]
Sorry just rebasing this, will re-open.

**Comment by sandeepmistry** [comment]
I've rebased this as https://github.com/sandeepmistry/arduino-nRF5/issues/96.

**Pull request closed without merging**